plugins/leemons-plugin-tests/frontend/src/pages/private/questions-banks/components/question-types/MonoResponse/components/ListInputRender.js

```diff
@@ -22,9 +22,16 @@ export function ListInputRender({
   value,
   onCancel,
   scrollRef,
+  responsesSaved,
   ...props
 }) {
-  const [store, render] = useStore(value || { useButton: true });
+  const [store, render] = useStore(value);
+  const [useButton, setUseButton] = React.useState(!value);
+
+  useEffect(() => {
+    const isValueSaved = responsesSaved?.some((response) => response.value === value);
+    setUseButton(!isValueSaved);
+  }, []);
```

Comment:
Won't be better:

```js
const [useButton] = useState(() => responsesSaved?.some((response) => response.value === value));
```

It will only run the check in the first render as `useEffect(noop, [])` as it's lazy, but it's a one liner plus avoiding a re-render due to setState.

Another option, would be using a useMemo, but less compatible with future versions of react

```js
const useButton = useMemo(() => responsesSaved?.some((response) => response.value === value), []);
```


For more reference: https://tkdodo.eu/blog/use-state-for-one-time-initializations